src/main/java/frc/robot/config/DynamicRobotConfig.java

```diff
@@ -0,0 +1,100 @@
+package frc.robot.config;
+
+import edu.wpi.first.wpilibj.Preferences;
+import java.util.logging.Logger;
+
+/**
+ * The DynamicRobotConfig class keeps track of periodically changing constants on the robot
+ *
+ * <p>This should only have offsets that would change with a mechanical change. There should be no
+ * IDs, Conversions, or setpoint information in this config.
+ */
+public class DynamicRobotConfig {
+  private static DynamicRobotConfig dynRobotConfig;
+
+  // Keys to load values from, shall not be modified under any circumstance
+  private static final String frontLeftOffset_key = "front Left Offset";
+  private static final String frontRightOffset_key = "front Right Offset";
+  private static final String backRightOffset_key = "back Right Offset";
+  private static final String backLeftOffset_key = "back Left Offset";
+
+  // Contains every key that is used
+  // if one is missing it won't be correctly saved, or init
+  private static final String[] allKeys = {
+    frontLeftOffset_key, frontRightOffset_key, backRightOffset_key, backLeftOffset_key
+  };
+
+  public final double frontLeftOffset;
+  public final double frontRightOffset;
+  public final double backRightOffset;
+  public final double backLeftOffset;
+
+  public TunerConstants getTunerConstants() {
+    return new TunerConstants(frontLeftOffset, frontRightOffset, backLeftOffset, backRightOffset);
+  }
+
+  private DynamicRobotConfig() {
+    boolean initNT = false;
+
+    if (Preferences.containsKey(frontLeftOffset_key)) {
+      frontLeftOffset = Preferences.getDouble(frontLeftOffset_key, 0d);
+    } else {
+      raiseWarning("Front Left Pod Offset NOT FOUND!! using default");
+      frontLeftOffset = 0;
+      initNT = true;
+    }
+
+    if (Preferences.containsKey(frontRightOffset_key)) {
+      frontRightOffset = Preferences.getDouble(frontRightOffset_key, 0d);
+    } else {
+      raiseWarning("Front Right Pod Offset NOT FOUND!! using default");
+      frontRightOffset = 0;
+      initNT = true;
+    }
+
+    if (Preferences.containsKey(backRightOffset_key)) {
+      backRightOffset = Preferences.getDouble(backRightOffset_key, 0d);
+    } else {
+      raiseWarning("Back Right Pod Offset NOT FOUND!! using default");
+      backRightOffset = 0;
+      initNT = true;
+    }
+
+    if (Preferences.containsKey(backLeftOffset_key)) {
+      backLeftOffset = Preferences.getDouble(backLeftOffset_key, 0d);
+    } else {
+      raiseWarning("Back Left Pod Offset NOT FOUND!! using default");
+      backLeftOffset = 0;
+      initNT = true;
+    }
```

Comment:
I considered using a dictionary but went against it for 3 reasons.
1. It would require all the preferences to have to the same type. (As in disallowing a String to be saved for a Limelight name)
2. The implementation for the configurability I wanted didn't seem cleaner. As in a function that takes in a error message, a key, and be able to raise a flag would be more difficult to read while sticking with common java types.
3. The conversion from a key to field would be difficult. then would also prohibit readability (because reflection code is confusing)